shm/src/Store.cpp

```diff
@@ -44,27 +44,38 @@ Store::Response Store::buy(Cargo* cargo, size_t amount, Player* player) {
     if (player->getAvailableSpace() < amount) {
         return Response::lack_of_space;
     }
+    player->buy(cargo, amount, cargo->getPrice());
+    auto it = std::find(begin(cargo_), end(cargo_), cargo);
+    if (it != end(cargo_)) {
+        (*it)->setAmount((*it)->getAmount() - amount);
+    }
+    // TO DO (DELETE CARGO FROM VECTOR - UNLOAD)
     return Response::done;   
 }
 
-Store::Response Store::sell(Cargo* cargo, size_t amount, Player* player) {
+Store::Response Store::sell(std::shared_ptr<Cargo> cargo, size_t amount, Player* player) {
     if (!cargo) {
         return Response::lack_of_cargo;
     }
     if (cargo->getAmount() + amount > STORE_CAPACITY) {
         return Response::lack_of_space;
     }
+    player->sell(cargo, amount, cargo->getPrice());
+    auto it = std::find(begin(cargo_), end(cargo_), cargo);
+    if (it != end(cargo_)) {
+        (*it)->setAmount((*it)->getAmount() + amount);
+    }
```

Comment:
probably it's not a good place to calculate amounts - it should be delegated to Ship class (unload method)